distributed/cli/tests/test_cli_utils.py

```diff
@@ -17,6 +17,7 @@ def test_uri_from_host_port():
     assert f('', 456, 123) == 'tcp://:456'
     assert f('', None, 123) == 'tcp://:123'
     assert f('', None, 0) == 'tcp://'
+    assert f('', 0, 123) == 'tcp://:0'
```

Comment:
Is this `Scheduler.start()`? If you are passing it `tcp://`, it binds to a random port:
```
>>> from distributed import Scheduler
>>> s = Scheduler()
>>> s.start('tcp://')
distributed.scheduler - INFO -   Scheduler at:  tcp://192.168.1.68:39230
<tornado.concurrent.Future at 0x7f4272531160>
>>> s.listen_address
'tcp://0.0.0.0:39230'
>>> s.port
39230
>>> s.address
'tcp://192.168.1.68:39230'
```